src/frontend/ui/ui_manager.rs

```diff
@@ -38,6 +38,7 @@ impl<'a> CanvasDrawable for CanvasDrawableStrategy<'a> {
     }
 }
 
+#[derive(Debug)]
```

Comment:
Won't work as our observers don't implement `Debug`. You might want to implement Debug yourself:

```rust
impl std::fmt::Debug for UiManager {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        f.debug_struct("UiManager")
            .field("tool_manager", &self.tool_manager)
            .field("canvas", &self.canvas)
            .field("selection", &self.selection)
            .field("drawables", &self.drawables)
            .field("app_mode_sender", &self.app_model_sender)
            .field("render_observerr", &"<closures>")
            .finish_non_exhaustive()
    }
}
```
Note that forcing our observers to implement Debug is not an option as every closure has it's unique type thus we have to implement debug for each closure individually.

A better option might be to use custom macros like this one: https://crates.io/crates/derivative

Reply:
Good point. I chose [derive_where](https://crates.io/crates/derive-where) since it seems to be maintained.